lib/compat.php

```diff
@@ -100,22 +100,43 @@ function gutenberg_add_rest_nonce_to_heartbeat_response_headers( $response ) {
 add_filter( 'wp_refresh_nonces', 'gutenberg_add_rest_nonce_to_heartbeat_response_headers' );
 
 /**
- * As a substitute for the default content `wpautop` filter, applies autop
- * behavior only for posts where content does not contain blocks.
+ * Ensures `wpautop` is only applied to non-block content.
  *
  * @param  string $content Post content.
- * @return string          Paragraph-converted text if non-block content.
+ * @return string          The unmodified content.
  */
 function gutenberg_wpautop( $content ) {
-	if ( has_blocks( $content ) ) {
-		return $content;
+	// If there are blocks in this content, we shouldn't run wpautop() on it later.
```

Comment:
There's no good reason.

I may have made a mental mistake and avoided `do_blocks` because it is only defined conditionally by this plugin, but actually, we only need to worry about avoiding wpautop when we define `do_blocks`. Thanks for asking.